backoffice/views.py

```diff
@@ -305,3 +306,49 @@ def vehicles(request):
     context["page_range"] = page_range
 
     return render(request, "backoffice/vehicles.html", context)
+
+
+@require_backoffice_access
+def lock(request, id):
+    context = {
+        "user": request.user,
+    }
+
+    vehicle = Vehicle.objects.get(pk=id)
+    context["vehicle"] = vehicle.name
+    perform_box_action(
+        request=request, vehicle=vehicle, action_to_perform="lock", user=request.user
+    )
+    return redirect(reverse("backoffice_vehicles"))
+
+
+@require_backoffice_access
+def unlock(request, id):
+    context = {
+        "user": request.user,
+    }
+
+    vehicle = Vehicle.objects.get(pk=id)
+    context["vehicle"] = vehicle.name
+    perform_box_action(
+        request=request, vehicle=vehicle, action_to_perform="unlock", user=request.user
+    )
+    return redirect(reverse("backoffice_vehicles"))
+
+
+def perform_box_action(request, vehicle, action_to_perform, user):
+    box_id = vehicle.box
+    time_to_expire = timezone.now() + timezone.timedelta(minutes=10)
+    action = BoxAction(
+        action=action_to_perform,
+        created_at=timezone.now(),
+        expires_at=time_to_expire,
+        box=box_id,
+        user_id=user.id,
+    )
+    action.save()
+    # message is printed and dispatched regardless of outcome - may be worth exploring options to send different messages
```

Comment:
```suggestion
    # FIXME: should send an error message here not a success message if something goes wrong.
```

At the very least, let's have a `FIXME` here so it shows up in greps for that. but might be better to actually handle the error case and do an "error" message instead of a "success" one in that case.

Reply:
In order to implement error messages it will require using `signals` or custom model methods, both will likely increase maintenance. There should not be an error in this situation unless there is a larger problem with the DB.